### DISCUSSION TITLE
How to access signal data when generators are failing?

### DISCUSSION THREAD

**Discussion by monoboy-hh** [danielhrisca/asammdf#696] [Q&A] [answered]
Python 3.7.6
Name: asammdf
Version: 7.0.7
Linux 5.4.0-109-generic

Hi danielhrisca,
first off thanks for this great lib. I see you are rather active over here so I thought I give it shot:

Using the iter_groups and iter_channels generators I can only access two of three available channels in my mf4 (variable name 'mdf' in the following). The third one raises an exception "wrong data type "11" for vlsd channel"
`for g in mdf.iter_groups():
     print(f'Channels in group: {g.keys()}')`
output:
Channels in group: Index(['GPSTimestamp', 'MotorSpeed'], dtype='object')
Ignoring **wrong data type "11" for vlsd channel,** mdf file appears to be invalid.
 
The mdf file was recorded with some vector logger and the vector validator claims it is legal. Despite the file being valid or not, it comes from a customer so there's not much to do about it.
However, the "missing" third channel is present in the mdf.channels_db, so I can access it through
`third_channel: Channel = mdf.groups[my_group_index].channels[my_channel_index]`
and can read the pointer to the Signal Data through
`sig_addr = third_channel.data_block_addr`

At this point I'm somewhat stuck how to finally retrieve pandas dataframes and I have the feeling it's not the right path anyway.
(Using something like
`stream = mdf.select(channels=['PointCloud'])`
fails with the same exception)

So my questions are:
- is there a way to work around the 'wrong data type exception'?
- or, how to 'manually' follow the pointers to the signal data to retrieve pandas df?

I can provide demo data, though about 250 MB in size.
Please find the structure attached as screenshot.

Thank you so much!
Tim

![demo](https://user-images.githubusercontent.com/59950865/166682115-b432279a-b7b0-4824-a334-6a0f9a2a2bc3.png)

**Comment by danielhrisca** [comment, marked as answer]
@monoboy-hh please try the development branch code

**Reply by monoboy-hh** [reply]
Wow, thanks for the ultra-quick fix, Daniel!
__version__ = "7.1.0.dev42" does the job

`Channels in group: Index(['GPSTimestamp', 'MotorSpeed'], dtype='object')
Channels in group: Index(['PointCloud'], dtype='object')`

Thank you!
Best,
Tim

**Comment by YoBruce45** [comment]
hello @danielhrisca 
from my side as well thank you for this very helpful library. 
I am facing kind of the same issue as @monoboy-hh : 

I open a `mdf_file ` and iter through the groups with
`for group in mdf_file.iter_groups()`

This fails though, as one group cannot be parsed, with the exception "_asammdf.blocks.utils.MdfException: wrong data type "0" for vlsd channel "ETH_Frame.DataBytes"_" and  I am not able to load the file at all. 

Is there a way to silently ignore this kind of exception and if yes, how?
(if the fix above applies, how can i use version = "7.1.0.dev42" ?)

Thanks a lot in advance for your support and kind regards

**Reply by danielhrisca** [reply]
can you send your file? I think there is something weird about it

**Comment by danielhrisca** [comment]
```python
for i in range(len(mdf_file.groups)):
    try:
        df = mdf_file.get_group(i)
    except Exception as e:
        print(e)
```

**Reply by YoBruce45** [reply]
thanks a lot, works like a charm =)
